Update triage and prioritization processes (#4852)
vmware-tanzu/kubeapps: site/content/docs/latest/reference/developer/issue-triage-process.md

| [Feature request](https://github.com/vmware-tanzu/kubeapps/issues/new?assignees=&labels=kind%2Fproposal&template=feature-request.md&title=) | File a request for a new feature.                 |
| [Support request](https://github.com/vmware-tanzu/kubeapps/issues/new?assignees=&labels=kind%2Fquestion&template=support-request.md&title=) | File a request for questions and support.         |
| [Issue](https://github.com/vmware-tanzu/kubeapps/issues/new?assignees=&labels=&template=issue.md&title=)                                    | File an issue to describe work to be performed.   |

Kubeapps keeps a backlog of issues on GitHub submitted both by maintainers and contributors.

- The backlog is completely open. The maintainer team, along with the entire Kubeapps community, files all feature enhancements, bugs, and potential future work into the open repository.
- Issues are closed if solved, or outdated (meaning the issue does not apply anymore according to the evolution of Kubeapps or it was waiting for more information which was never received).

## 🏷 Labeling

> Kubeapps mantainers team operates with 4 groups of labels:

Reviewer: ```suggestion
Kubeapps maintainers team operates with 4 groups of labels:
```

### `kind/*`:

The type of issue. kind powers our filtering to understand what qualifies as a bug, proposal, feature, enhancement, question or documentation.

| Label                | Description                                                      |
| -------------------- | ---------------------------------------------------------------- |
| `kind/bug`           | An issue that reports a defect in an existing feature            |
| `kind/documentation` | An issue that reports an update related to project documentation |
| `kind/proposal`      | An issue that reports a new feature proposal to be discussed     |
| `kind/feature`       | An issue that reports a feature (approved) to be implemented     |
| `kind/enhancement`   | An issue that reports an enhancement for an implemented feature  |

### `component/*`:

> The area of the issue. Components are high level areas of Kubeapps architecture. They are used to group issues into smaller and related parts.

Reviewer 2: ```suggestion
The relevant component(s) for the issue. Components are high level areas of the Kubeapps architecture. They are used to group issues together with other related issues.
```

| Label                        | Description                                                                                      |
| ---------------------------- | ------------------------------------------------------------------------------------------------ |
| `component/api-server`       | An issue related to kubeapps api-server                                                          |
| `component/apprepository`    | An issue related to kubeapps apprepository                                                       |
| `component/asset-syncer`     | An issue related to kubeapps asset-syncer (to be deprecated)                                     |
| `component/assetsvc`         | An issue related to kubeapps assetsvc (to be deprecated)                                         |
| `component/authentication`   | An issue related to kubeapps authentication                                                      |
| `component/ci`               | An issue related to kubeapps ci system                                                           |
| `component/kubeops`          | An issue related to kubeops (to be deprecated)                                                   |

> | `component/packages`         | An issue related to kubeapps packging formats to be distributed (Helm chart and Carvel packages) |

Reviewer: ```suggestion
| `component/packages`         | An issue related to kubeapps packaging formats to be distributed (Helm chart and Carvel packages) |
```

| ------------------------ | -------------------------------------------------------------------------------- |
| `awaiting-more-evidence` | Need more info requested to actually get it done.                                |
| `cla-notrequired`        | Automatic label for CLA signature                                                |
| `cla-rejected`           | Automatic label for CLA signature when rejected                                  |
| `dependencies`           | Automatic label set to pull requests that update a dependency file               |
| `github_actions`         | Label assigned to pull requests that update GitHub Actions code                  |
| `go`                     | Automatic label set to pull requests that update Go code                         |
| `good first issue`       | Good first issues to start contributing to Kubeapps.                             |
| `help wanted`            | The maintainer team wants help on an issue or pull request.                      |
| `javascript`             | Automatic label set to pull requests that update javascript code                 |

> | `next-iteration`         | Label to mark issues to be discussed in next planning session                    |

Reviewer: ```suggestion
| `next-iteration`         | Label to mark issues to be discussed in the next planning session                    |
```

| 'L'   | A story that requires investigation and possibly will take a person a full 2-week iteration |
| 'XL'  | A story too big or with too many unknowns. Needs investigation and split into several ones  |

| priority/ |       |       |       |
| --------- | ----- | ----- | ----- |

> | ⛔️ P0    | 🔴 P1 | 🟠 P2 | 🔵 P3 |

Reviewer 2: Might be worth a small description for each:

- P0 - drop everything and fix it
- P1 - ...

Fine either way.
